### PR TITLE
Upgrade DatabaseConnector & use CohortGenerator for read/write csv operations

### PR DESCRIPTION
**Pull request by anthonysena**
I'd suggest that this PR is merged into a feature branch, edited and then considered for merging into main. 

This PR aims to update the version of DatabaseConnector in the `renv.lock` file used by the module based on https://github.com/OHDSI/Strategus/issues/16. It also makes use of CohortGenerator for read/write csv operations.

This branch also includes some changes that should be revised before merging into main. I'll detail those in the specific code blocks in this PR.

### CODE REVIEW

File: MetaData.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "Name": "CohortMethodModule",
-  "Version": "0.0.2",
+  "Version": "0.0.4",
```
Comment (anthonysena) [inline MetaData.json:3]: This should reflect 0.0.3 and not 0.0.4 from my testing.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+CohortMethodModule 0.0.3
+========================
+
+Update to use DatabaseConnector v5.0.4
```
Comment (anthonysena) [inline NEWS.md:4]: This comment is incomplete and should also mention the use of CohortGenerator

File: SettingsFunctions.R [hunk 3]
```diff
@@ -12,9 +12,9 @@ createCohortMethodModuleSpecifications <- function(cmAnalysisList,
   }
 
   specifications <- list(module = "CohortMethodModule",
-                         version = "0.0.2",
+                         version = "0.0.4",
                          remoteRepo = "github.com",
-                         remoteUsername = "ohdsi",
+                         remoteUsername = "anthonysena",
```
Comment (anthonysena) [inline SettingsFunctions.R:17]: This should be rejected; instead the version should be 0.0.3 and remoteUsername = 'ohdsi'

File: extras/ModuleMaintenance.R [hunk 4]
```diff
@@ -44,7 +44,7 @@ rCode[grep("return\\(analysis\\)", rCode)]  <- sprintf("
   specifications <- list(module = \"%s\",
                          version = \"%s\",
                          remoteRepo = \"github.com\",
-                         remoteUsername = \"ohdsi\",
+                         remoteUsername = \"anthonysena\",
```
Comment (anthonysena) [inline extras/ModuleMaintenance.R:47]: `remoteUsername = \"ohdsi\"` should be restored

**Comment by schuemie** [comment]
I'm sorry, I'm a bit confused. You want me to accept this pull request or not?

**Comment by anthonysena** [comment]
Let me clean up the branch on my fork and then this should be ready to merge. Sorry for the confusion.

**Comment by anthonysena** [comment]
OK this is now ready for merging.

**Pull request merged**